code/modules/antagonists/heretic/items/forbidden_book.dm

```diff
@@ -48,18 +54,22 @@
 		heretic_datum.try_draw_rune(user, target, drawing_time = 12 SECONDS)
 		return TRUE
 
-/*
- * Plays a little animation that shows the book opening and closing.
- */
+/// Plays a little animation that shows the book opening and closing.
 /obj/item/codex_cicatrix/proc/open_animation()
+	if(book_open == TRUE)
+		return
+
 	icon_state = "[base_icon_state]_open"
 	flick("[base_icon_state]_opening", src)
+	book_open = TRUE
 
 	addtimer(CALLBACK(src, PROC_REF(close_animation)), 5 SECONDS)
 
-/*
- * Plays a closing animation and resets the icon state.
- */
+/// Plays a closing animation and resets the icon state.
 /obj/item/codex_cicatrix/proc/close_animation()
+	if(book_open == FALSE)
```

Comment:
Your suggestions just don't work and if i change switch to if the timer of closing up won't reset.

Reply:
Your suggestion don't work an i change `switch` to `if` the timer of closing up won't reset upon closing and opening again.

Reply:
melbert accidentally put `close_animation` in both cases in their suggestion, did you fix that in your implementation? i don't see why it shouldn't work otherwise

Reply:
Opening a book has a timer for 5 seconds that closes the book and without switch it doesn't override the timer. 
I'll recheck it to be 100% sure.

Reply:
https://user-images.githubusercontent.com/93882977/232307074-99c4faf3-07ae-49b8-af7a-aea70511e7f5.mp4
Rechecked